File: assignment-18/task3.cpp
#include <iostream>

int factorial(int n) {
    if (n == 0) return 1;
    return n * factorial(n - 1);
}

int main() {
    std::cout << "Input: 5 → Output: Factorial = " << factorial(5) << std::endl;
    std::cout << "Input: 0 → Output: Factorial = " << factorial(0) << std::endl;
    return 0;
}

File: assignment-18/task3.py
def factorial(n):
    """Return factorial of n using recursion."""
    if n == 0:
        return 1
    return n * factorial(n - 1)

if __name__ == "__main__":
    # Calls as requested
    print(f"Input: 5 → Output: Factorial = {factorial(5)}")
    print(f"Input: 0 → Output: Factorial = {factorial(0)}")
